opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp

```diff
@@ -51,7 +55,30 @@ BlackoilAquiferModel<TypeTag>::initFromRestart(const data::Aquifers& aquiferSoln
 template <typename TypeTag>
 void
 BlackoilAquiferModel<TypeTag>::beginEpisode()
-{}
+{
+    // TODO: not totally sure this is the function should be used.
+    // basically, we want to update the aquifer related information from SCHEDULE setup in this section
+    // it is the beginning of a report step
+
+    const auto& connections = this->simulator_.vanguard().eclState().aquifer().connections();
+    const int report_step  = this->simulator_.episodeIndex();
+    const auto& aqufluxs = this->simulator_.vanguard().schedule()[report_step].aqufluxs;// .aquflu// simulator.vanguard().schedule()[reportStepIdx].events()
+    for (const auto& elem : aqufluxs) {
+        const int id = elem.first;
+        auto find = std::find_if(begin(this->aquifers), end(this->aquifers), [id](auto& v){ return v->aquiferID() == id; });
+        if (find == this->aquifers.end()) {
+            // the aquifer id does not exist in aquifers yet
+            const auto& aquinfo = elem.second;
+            auto aqf = std::make_unique<AquiferConstantFlux<TypeTag>>(aquinfo, connections.getConnections(aquinfo->id), this->simulator_);
+            this->aquifers.push_back(std::move(aqf));
+        } else {
+            const double prev_cumulative_flux = (*find)->cumulativeFlux();
+            const auto& aquinfo = elem.second;
+            auto aqf = std::make_unique<AquiferConstantFlux<TypeTag>>(aquinfo, connections.getConnections(aquinfo->id), this->simulator_, prev_cumulative_flux);
```

Comment:
For AQUFLUX, it might be fine to do it this way, especially now, we only use it to update the flux rate using the record of AQUFLUX. But for AQUFETP , when we specify AQUFETP for a certain report step,  there will be a re-calculation of the aquifer pressure, it is important to know the report step that the AQUFETP keyword is actually specified.  (not tested with AQUCT, while might be even more complicated besides the same problem with AQUFETP). 

It is a problem related how we do the Schedule with ` std::vector<ScheduleState> snapshots;`, with which we keep the same ScheduleState for report steps until some keywords are specified to change it.

Reply:
Yes, thanks. That is also what I thought.